src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1124,48 +1124,71 @@ public string Replace(char oldChar, char newChar)
 
         public string Replace(string oldValue, string? newValue)
         {
-            if (oldValue == null)
+            if (oldValue is null)
+            {
                 throw new ArgumentNullException(nameof(oldValue));
+            }
             if (oldValue.Length == 0)
+            {
                 throw new ArgumentException(SR.Argument_StringZeroLength, nameof(oldValue));
+            }
 
-            // Api behavior: if newValue is null, instances of oldValue are to be removed.
-            newValue ??= string.Empty;
+            // If newValue is null, treat it as an empty string.  Callers use this to remove the oldValue.
+            newValue ??= Empty;
 
+            // Track the locations of oldValue to be replaced.
             var replacementIndices = new ValueListBuilder<int>(stackalloc int[StackallocIntBufferSizeLimit]);
```

Comment:
always (stack)allocated even if we go `Replace(oldValue[0], newValue[0]);` fast path

Reply:
Is that a problem?

Reply:
However, it's probably still better than putting two `stackalloc int[]` in those two branches 🤔

Reply:
Since the assembly is marked `[SkipLocalsInit]` this shouldn't result in significant overhead, right? Would need to crack open the codegen to see if anything changed.

Reply:
> Since the assembly is marked [SkipLocalsInit] this shouldn't result in significant overhead, right?

Right.